source/updateCheck.py

```diff
@@ -242,17 +283,78 @@ def onDownloadButton(self, evt):
 		DonateRequestDialog(gui.mainFrame, self._download)
 
 	def _download(self):
-		if self.isInstalled:
-			UpdateDownloader(self.urls, fileHash=self.fileHash).start()
-		else:
-			os.startfile(self.urls[0])
+		UpdateDownloader(self.urls, fileHash=self.fileHash).start()
 		self.Destroy()
 
 	def onLaterButton(self, evt):
 		state["dontRemindVersion"] = None
 		saveState()
 		self.Close()
 
+class UpdateAskInstallDialog(wx.Dialog):
+
+	def __init__(self, parent, destPath, version):
+		self.destPath=destPath
+		self.version = version
+		storeUpdatesDirWritable=os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir)
+		# Translators: The title of the dialog asking the user to Install an NVDA update.
+		super(UpdateAskInstallDialog, self).__init__(parent, title=_("NVDA Update"))
+		mainSizer = wx.BoxSizer(wx.VERTICAL)
+		if not storeUpdatesDirWritable:
+			# Translators: A message indicating that an updated version of NVDA can either be installed or discarded.
+			message=_("You can either install it now, or discard the update and download it again at a later time.")
+		else:
+			# Translators: A message indicating that an updated version of NVDA can be installed, postponed or discarded.
+			message=_("You can install it now, or install it later from the NVDA menu or the exit dialog, if enabled.\n")
+		mainSizer.Add(wx.StaticText(self, label=
+			# Translators: A message indicating that an updated version of NVDA is ready to be installed.
+			_("Update is ready to install.\n")+message+
+			# Translators: A message telling the user that it is possible to discard a downloaded update.
+			_("Alternatively, you can discard the update and download it again at a later time.")
+		))
+		# Translators: The label of a button to install an NVDA update.
+		item = wx.Button(self, label=_("&Install update"))
+		item.Bind(wx.EVT_BUTTON, lambda evt: executeUpdate())
+		item.SetFocus()
+		mainSizer.Add(item)
+		if storeUpdatesDirWritable:
+			# Translators: The label of a button to postpone an NVDA update.
+			item = wx.Button(self, label=_("&Postpone update"))
+			item.Bind(wx.EVT_BUTTON, self.onPostponeButton)
+			mainSizer.Add(item)
+		# Translators: The label of a button to discard an NVDA update.
+		item = wx.Button(self, wx.ID_CLOSE, label=_("&Discard update"))
+		item.Bind(wx.EVT_BUTTON, lambda evt: self.Close())
+		sizer.Add(item)
+		self.Bind(wx.EVT_CLOSE, self.onDiscard)
+		self.EscapeId = wx.ID_CLOSE
+		self.Sizer = mainSizer
+		mainSizer.Fit(self)
+		self.Center(wx.BOTH | wx.CENTER_ON_SCREEN)
+		self.Show()
+
+	def onPostponeButton(self, evt):
+		finalDest=os.path.join(storeUpdatesDir, os.path.basename(self.destPath))
+		try:
+			os.renames(self.destPath, finalDest)
+		except:
+			gui.messageBox(
+				# Translators: The message when a downloaded update file could not be preserved.
+				_("The downloaded file could not be postponed properly."),
+				_("Error"), wx.OK | wx.ICON_ERROR)

```

Comment:
This is missing a translator comment. I think there is an automated test that would catch this, you can run this with `scons checkPot`